ext/vscode/src/extension.ts

```diff
@@ -0,0 +1,62 @@
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as os from 'os';
+import { execSync } from 'child_process';
+
+export function activate(context: vscode.ExtensionContext) {
+    const didChangeEmitter = new vscode.EventEmitter<void>();
+
+    context.subscriptions.push(
+        vscode.lm.registerMcpServerDefinitionProvider('azureMcpProvider', {
+            onDidChangeMcpServerDefinitions: didChangeEmitter.event,
+            provideMcpServerDefinitions: async () => {
+                let npmPkg = '';
+                let binary = '';
+                if (process.platform === 'win32') {
+                    npmPkg = '@azure-mcp/server-win-x64';
+                    binary = 'azmcp.exe';
+                } else if (process.platform === 'darwin') {
+                    npmPkg = '@azure-mcp/server-osx-x64';
+                    binary = 'azmcp';
```

Comment:
Not sure if it's the case here. In our extension, we have seen that the executable files in our vsix are by default created without the execute permission on mac and linux. You may need to configure the file permission for the bundled executables.